generators/ionic/resources/oauth2/cypress/support/oauth2.ts

```diff
@@ -0,0 +1,179 @@
+/* eslint-disable @typescript-eslint/no-namespace, @typescript-eslint/naming-convention */
+// eslint-disable-next-line spaced-comment
+/// <reference types="cypress" />
+import { apiHost } from './config';
+import { environment } from '../../src/environments/environment';
+
+const {
+  oidcConfig: { redirect_url: redirect_uri, scopes: scope, audience },
+} = environment;
+
+// Get oauth2 basic data
+const getOauth2Data = () =>
+  cy
+    .request({
+      url: `${apiHost}/api/auth-info`,
+      followRedirect: false,
+    })
+    .then(({ body: info }) => {
+      const { issuer, clientId } = info;
+      return cy
+        .request({
+          url: `${issuer}/.well-known/openid-configuration`,
+          followRedirect: false,
+        })
+        .then(({ body: configuration }) => ({
+          info,
+          configuration,
+          qs: {
+            redirect_uri,
+            client_id: clientId,
+            response_type: 'code',
+            scope,
+            audience,
+          },
+        }));
+    });
+
+// Get authorization from keycloak
+const keycloakLogin = ({ configuration, qs }, username: string, password: string) => {
+  const { authorization_endpoint } = configuration;
+  return cy
+    .request({
+      url: authorization_endpoint,
+      qs,
+      followRedirect: false,
+    })
+    .then(response => {
+      const html = document.createElement('html');
+      html.innerHTML = response.body;
+
+      const form = html.getElementsByTagName('form')[0];
+      const url = form.action;
+      return cy.request({
+        method: 'POST',
+        url,
+        followRedirect: false,
+        form: true,
+        body: {
+          username,
+          password,
+        },
+      });
+    });
+};
+
+// Get authorization from auth0
+const auth0Login = ({ configuration, qs }, username: string, password: string) => {
+  const { authorization_endpoint } = configuration;
+  return cy
+    .request({
+      url: authorization_endpoint,
+      qs,
+      followRedirect: true,
+    })
+    .then(response => {
+      const html = document.createElement('html');
+      html.innerHTML = response.body;
+      const state = html.querySelector('input[name="state"]').value;
+      return cy.request({
+        method: 'POST',
+        url: `${new URL(authorization_endpoint).origin}/u/login`,
+        followRedirect: true,
+        form: true,
+        body: {
+          state,
+          action: 'default',
+          username,
+          password,
+        },
+      });
+    });
```

Comment:
I get a 404 on this call with Auth0.

<img width="1241" alt="Screen Shot 2022-04-28 at 19 50 16" src="https://user-images.githubusercontent.com/17892/165874391-204fa888-3937-48f7-8e8b-002f840d183e.png">